lib/plugins/aws/package/compile/events/lib/ensureApiGatewayCloudWatchRole.js

```diff
@@ -1,13 +1,30 @@
 'use strict';
 
 const { memoize } = require('lodash');
+const BbPromise = require('bluebird');
 const { addCustomResourceToService } = require('../../../../customResources');
 
 module.exports = memoize(provider => {
   const cfTemplate = provider.serverless.service.provider.compiledCloudFormationTemplate;
+  const logicalId = provider.naming.getApiGatewayAccountCloudWatchRoleResourceLogicalId();
   const customResourceLogicalId = provider.naming.getCustomResourceApiGatewayAccountCloudWatchRoleResourceLogicalId();
   const customResourceFunctionLogicalId = provider.naming.getCustomResourceApiGatewayAccountCloudWatchRoleHandlerFunctionLogicalId();
 
+  // There may be a specific role ARN provided in the configuration
+  const config = provider.serverless.service.provider;
+  const restApi = config.logs && config.logs.restApi;
+  const configuredRoleArn = restApi && restApi.role;
+
+  if (configuredRoleArn) {
+    cfTemplate.Resources[logicalId] = {
+      Type: 'AWS::ApiGateway::Account',
+      Properties: {
+        CloudWatchRoleArn: configuredRoleArn,
+      },
+    };
+    return BbPromise.resolve(cfTemplate);
```

Comment:
It'll be better to wrap whole body of a function with `BbPromise.try` (it'll make future convertion to async/await easier)